src/core/lib/event_engine/extensions/transport_extension.h

```diff
@@ -0,0 +1,48 @@
+// Copyright 2024 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GRPC_SRC_CORE_LIB_EVENT_ENGINE_EXTENSIONS_TRANSPORT_EXTENSION_H
+#define GRPC_SRC_CORE_LIB_EVENT_ENGINE_EXTENSIONS_TRANSPORT_EXTENSION_H
+
+#include "absl/functional/any_invocable.h"
+#include "absl/strings/string_view.h"
+
+#include <grpc/support/port_platform.h>
+
+#include "src/core/lib/slice/slice_buffer.h"
+
+namespace grpc_event_engine {
+namespace experimental {
+
+/// An Endpoint extension class that will be supported by EventEngine endpoints
+/// which can send data to a transport and receive data from it.
+class TransportExtension {
+ public:
+  virtual ~TransportExtension() = default;
+  static absl::string_view EndpointExtensionName() {
+    return "io.grpc.event_engine.extension.transport_extension";
+  }
+
+  /// Sends data to transport through `cb`.
+  virtual void SetSendDataCallback(
```

Comment:
Let's call this `SetSendFrameCallback` to make it explicit that this is sending complete frames/datagrams.

Reply:
Done.